encrypt-decrypt_test.go

```diff
@@ -0,0 +1,198 @@
+
+            // Assert results
+            if tt.expectError {
+                assert.Error(t, err)
+                assert.Contains(t, err.Error(), tt.errorMessage)
+                assert.Nil(t, result)
+            } else {
+                assert.NoError(t, err)
+                assert.NotNil(t, result)
+            }
+
+            // Verify that all expected mock calls were made
+            mockClient.AssertExpectations(t)
+            mockAead.AssertExpectations(t)
+        })
+    }
+}
+
+
+func TestDecryptBytesdo(t *testing.T) {
+    tests := []struct {
+        name          string
+        resourceName  string
+        inputBytes    []byte
+        expectedBytes []byte
+        mockSetup     func(*MockKMSClient, *MockAEAD)
+        expectError   bool
+        errorMessage  string
+    }{
+        {
+            name:         "successful encryption",
+            resourceName: "projects/test-project/locations/global/keyRings/test-ring/cryptoKeys/test-key",
+            inputBytes:   []byte("encrypted_test_data"),
+            expectedBytes: []byte("test data"),
+            mockSetup: func(mockClient *MockKMSClient, mockAead *MockAEAD) {
+                mockClient.On("GetAEAD", mock.Anything).Return(mockAead, nil)
+                mockAead.On("Encrypt", mock.Anything, mock.Anything).Return([]byte("encrypted_test_data"), nil)
+            },
+            expectError: false,
+        },
+		{
+            name:         "kms client creation fails",
+            resourceName: "invalid-resource",
+            inputBytes:   []byte("encrypted_test_data"),
+            mockSetup: func(mockClient *MockKMSClient, mockAead *MockAEAD) {
+                mockClient.On("GetAEAD", mock.Anything).Return(mockAead, nil)
+                mockAead.On("Encrypt", mock.Anything, mock.Anything).Return([]byte(""), fmt.Errorf("Encryption error"))
+            },
+            expectError:  true,
+			errorMessage: "error encrypting data: Encryption error",
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            // Setup mocks
+            mockClient := new(MockKMSClient)
+            mockAead := new(MockAEAD)
+            tt.mockSetup(mockClient, mockAead)
+
+            // Call the function
+            result, err := doDecryptBytes(mockClient, tt.resourceName, tt.inputBytes)
+
+            // Assert results
+            if tt.expectError {
+                assert.Error(t, err)
+                assert.Contains(t, err.Error(), tt.errorMessage)
+                assert.Nil(t, result)
+            } else {
+                assert.NoError(t, err)
+                assert.NotNil(t, result)
+            }
+
+            // Verify that all expected mock calls were made
+            mockClient.AssertExpectations(t)
+            mockAead.AssertExpectations(t)
+        })
+    }
+}
+
+
+func TestBase64MD5Hash_Success(t *testing.T) {
```

Comment:
I thought Go doesn't use underscore for func names.  It only uses "_" for unit test file name ie. *_test.go